pkg/storegateway/bucket_index_postings.go

```diff
@@ -363,6 +363,9 @@ func (selectAllStrategy) selectPostings(groups []postingGroup) (selected, omitte
 // `__name__="cpu_seconds_total"` selects 500K series, then the strategy calculates that the whole query will
 // select no more than 500K series. It uses this to calculate that in the worst case we will fetch 500K * tsdb.EstimatedSeriesP99Size
 // bytes for the series = 256 MB. So it will not fetch more than 256 MB of posting lists.
+//
+// We found that this strategy may cause increased API calls for cases where it omits the __name__ posting group.
+// Because of this now the strategy always selects the __name__ posting group regardless of its size.
```

Comment:
[nit] I suggest to avoid using temporal references (e.g. "now") to comments that may stick around for a long time.

```suggestion
// Because of this, the strategy always selects the __name__ posting group regardless of its size.
```